AppServer_Java/src/com/google/appengine/api/taskqueue/dev/AppScaleTaskQueueClient.java

```diff
@@ -135,8 +134,11 @@ private String getNginxPort()
 
     private String getAppId()
     {
-        String appId = System.getProperty("APPLICATION_ID");
-        return appId;
+    	if(this.appId == null)
+    	{
+    		this.appId = System.getProperty("APPLICATION_ID");
```

Comment:
Reading twice in a row has no value IMHO.
I think it's better to print a warning instead in this loop.